### PR TITLE
Fix macOS Platform, Custom URL Scheme Handler, Polish Demos.

### PR DESCRIPTION
**Pull request by chriszielinski**
* `DownView` did not work on macOS 10.14. It appears `loadHTMLString(_:baseURL:)` does not allow local file access, unlike on the iOS platform. On macOS, it now copies the necessary files to a temporary directory and uses `loadFileURL(_:allowingReadAccessTo:)`.
* Added support for custom URL scheme handlers for macOS 10.13+ and iOS 11.0+. As well as a test case for it.
* Added a macOS demo application and polished the iOS demo.

I'm unsure of the changelog formatting, so have not updated that.

### CODE REVIEW

**Comment by chriszielinski** [comment]
Travis is attempting to test `testCustomURLSchemeHandler()` on macOS 10.12.* even though it's denoted `@available(iOS 11.0, macOS 10.13, *)`. Not sure why... so I can either add an internal compiler directive to pass that case trivially, or update the` osx_image` Travis uses to a minimum of `xcode9.3`.

File: Down-Example/Down-Example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -363,17 +545,17 @@
...
+				DEVELOPMENT_TEAM = 9V456WSURS;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/Down-Example",
 				);
 				INFOPLIST_FILE = "Down-Example/Info.plist";
 				IPHONEOS_DEPLOYMENT_TARGET = 10.2;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = "com.down.Down-Example";
+				PRODUCT_BUNDLE_IDENTIFIER = "com.bigzlabs.Down-Example";
```
Comment (iwasrobbed) [inline Down-Example/Down-Example.xcodeproj/project.pbxproj]: Please leave bundle id's as they were (here and elsewhere)

File: Source/Views/DownView.swift [hunk 2]
```diff
@@ -87,29 +107,75 @@ private extension DownView {
...
 
     func htmlFromTemplate(_ htmlString: String) throws -> String {
-        let template = try NSString(contentsOf: baseURL, encoding: String.Encoding.utf8.rawValue)
+        let template = try String(contentsOf: baseURL, encoding: .utf8)
         return template.replacingOccurrences(of: "DOWN_HTML", with: htmlString)
     }
 
+    #if os(macOS)
+    func createTemporaryBundle(pageHTMLString: String) throws -> URL {
+        let bundleResourceURL = self.bundle.resourceURL!
```
Comment (iwasrobbed) [inline Source/Views/DownView.swift]: Please guard and throw instead of force unwrapping here

File: Tests/DownViewTests.swift [hunk 3]
```diff
@@ -95,6 +96,53 @@ class DownViewTests: XCTestCase {
             }
         }
     }
+
+    @available(iOS 11.0, macOS 10.13, *)
+    func testCustomURLSchemeHandler() {
```
Comment (iwasrobbed) [inline Tests/DownViewTests.swift]: Thanks for adding a test 💯

File: Down-Example/Down-Example/ViewController.swift [hunk 4]
```diff
@@ -11,66 +11,67 @@ import Down
...
+        DispatchQueue.global(qos: .userInteractive).async {
+            self.loadReadMe()
+        }
+    }
+
+    func loadReadMe() {
+        var readMeContents: String
+
+        do {
+            let readMeURL = URL(string: "https://raw.githubusercontent.com/iwasrobbed/Down/master/README.md")!
```
Comment (iwasrobbed) [inline Down-Example/Down-Example/ViewController.swift]: I would prefer to render local markdown here (so people don't think it's slow based upon their network connection). 

It looks like that was your first intention with `README-11:2:18.md` but I don't see it being used. Mind switching back to that?

The `progressIndicator` won't be necessary in that scenario.

File: Down-Example/macOS Demo/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        DispatchQueue.global(qos: .userInteractive).async {
+            self.loadReadMe()
+        }
+    }
+
+    func loadReadMe() {
+        var readMeContents: String
+
+        do {
+            let readMeURL = URL(string: "https://raw.githubusercontent.com/iwasrobbed/Down/master/README.md")!
```
Comment (iwasrobbed) [inline Down-Example/macOS Demo/ViewController.swift]: Same here re: rendering local markdown versus having a potentially slow network request and then eventually rendering.

File: Down-Example/Down-Example.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -249,6 +376,57 @@
...
+				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
+				CODE_SIGN_ENTITLEMENTS = "macOS Demo/macOS Demo.entitlements";
+				CODE_SIGN_IDENTITY = "Mac Developer";
+				CODE_SIGN_STYLE = Automatic;
+				COMBINE_HIDPI_IMAGES = YES;
+				DEVELOPMENT_TEAM = 9V456WSURS;
+				GCC_C_LANGUAGE_STANDARD = gnu11;
+				INFOPLIST_FILE = "macOS Demo/Info.plist";
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.14;
```
Comment (iwasrobbed) [inline Down-Example/Down-Example.xcodeproj/project.pbxproj]: Please set the deployment target to `10.11` to match the podspec

File: Down-Example/Down-Example/ViewController.swift (hunk 4 above)
Comment (chriszielinski) [reply Down-Example/Down-Example/ViewController.swift]: Ah, good catch. Both now use a shared local version.

**Comment by iwasrobbed** [comment]
Looks great, thanks again @chriszielinski !!

**Comment by iwasrobbed** [comment]
Released as v0.6.1 👍 🚀

**Pull request merged**